spec/lib/synapse/service_watcher_zookeeper_spec.rb

```diff
@@ -150,6 +150,20 @@
         expect(subject).to receive(:set_backends).with([],{})
         subject.send(:watcher_callback).call
       end
+
+      it 'does not crash if there is no zk node' do
```

Comment:
the better way to do this is:
```
context "when there is no zk node" do
  before :each do
    allow(..)
    expect(..)
  end
   ...
   it "does not crash" do
     ...
   end
end
```

This test is also very similar and has a lot of shared `expect` calls with the previous test (it also raises a `ZK::Exceptions::NoNode`) so maybe we can clean that up a bit (ex: put `generator_config_path` in a `let(..)` statement)